### ISSUE TITLE
[MU4 Issue] Closing the last score shouldn't close MU (if many scores were opened)

### ISSUE THREAD

**Issue by DmitryArefiev** [musescore/MuseScore#10399]
https://user-images.githubusercontent.com/90187801/151548181-4dbeabb6-08ea-4e66-a593-c32135628c21.mp4

**To Reproduce**
1. Open score A
2. Open score B 
3. Close score B (with X button)
4. Close score A (with X button) - it should not close MU 

**Platform information**
 - OS: Win10 (didn't try Mac, Lin)

**Comment by HemantAntony** [comment]
I investigated this and found out that this happens only if you close both instances quickly (at least in the current master). If you give a couple of seconds in between, this doesn't happen. If I had to guess I would say this happens because `multiInstancesProvider()->instances().size()` does not update quickly enough

**Comment by Tantacrul** [comment]
(apart from the crash that sometimes happens when you close scores)

This looks to be behaving correctly for me. @DmitryArefiev - can you check?

**Comment by DmitryArefiev** [comment]
@Tantacrul Still occur for me. Found one important step to reproduce - the second score you should open with Ctr+O (not from Home>Scores). Updated description


https://user-images.githubusercontent.com/90187801/171046846-5149a2fb-12ae-43bb-b366-e0cd438c61ea.mp4

**Comment by Tantacrul** [comment]
OK, I've checked again and I'm not getting this 😆 - and this time I really tried to follow your exact guidelines!

What do you think? Can we close this?

**Comment by aamri2** [comment]
I've found simpler conditions that seem to suggest it may be unrelated to the "opening multiple scores" factor:
https://user-images.githubusercontent.com/16510891/197375559-0662f020-6ee7-4ce2-a621-806150170103.mp4
If I open a score from a fresh window of MuseScore using the "Open other..." option and then close the file using the X on the score (not the window), the window remains open for a while, displaying the loading cursor, and then seems to crash. This occurs consistently and appears to be unexpected behaviour, and the same does not occur if I open a score using the recent scores window, or if I create a new score, and then close it.

**Comment by DmitryArefiev** [comment]
> OK, I've checked again and I'm not getting this 😆 - and this time I really tried to follow your exact guidelines @DmitryArefiev !
> 
> What do you think? Can we close this?




@Tantacrul The issue still occurs on my side. But I think we can move it to 4.x because there is no data loss in this bug.

https://user-images.githubusercontent.com/90187801/197407467-afdfe00c-8622-48d1-8696-3eb98896f9c8.mp4

**Comment by Tantacrul** [comment]
Hmm. I think it's pretty bad for it to be closing it completely. I'm tempted to move it to Release

**Comment by DmitryArefiev** [comment]
> I've found simpler conditions that seem to suggest it may be unrelated to the "opening multiple scores" factor: https://user-images.githubusercontent.com/16510891/197375559-0662f020-6ee7-4ce2-a621-806150170103.mp4 If I open a score from a fresh window of MuseScore using the "Open other..." option and then close the file using the X on the score (not the window), the window remains open for a while, displaying the loading cursor, and then seems to crash. This occurs consistently and appears to be unexpected behaviour, and the same does not occur if I open a score using the recent scores window, or if I create a new score, and then close it.

@aamri2 I think that's another bug. I was able reproduce your issue but with opened Mixer only. Logged #13933 for this.

**Comment by DmitryArefiev** [comment]
> Hmm. I think it's pretty bad for it to be closing it completely. I'm tempted to move it to Release

@Tantacrul Yeah. Other similar crash issues when closing score (but not the app) #13933 #12549

**Comment by Eism** [comment]
Also reproduced on Linux

**Comment by DmitryArefiev** [comment]
Fixed in #14007
